apps/main/src/dex/features/add-gauge-reward-token/ui/TokenSelector.tsx

```diff
@@ -24,23 +25,34 @@ export const TokenSelector = ({
   const network = useStore((state) => state.networks.networks[chainId])
   const rewardTokenId = watch('rewardTokenId')
   const { tokensMapper } = useTokensMapper(chainId)
+
   const { data: gaugeRewardsDistributors, isSuccess: isGaugeRewardsDistributorsSuccess } = useGaugeRewardsDistributors({
     chainId,
     poolId,
   })
 
   const filteredTokens = useMemo(() => {
     const gaugeRewardTokens = Object.keys(gaugeRewardsDistributors || {})
-    return Object.values(tokensMapper).filter(
-      (token): token is Token =>
-        token !== undefined &&
-        token.decimals === 18 &&
-        !aliasesCrv &&
-        ![...gaugeRewardTokens, zeroAddress, NETWORK_TOKEN, aliasesCrv].some((rewardToken) =>
-          isAddressEqual(rewardToken as Address, token.address as Address),
-        ),
-    )
-  }, [gaugeRewardsDistributors, tokensMapper, aliasesCrv])
+    return Object.values(tokensMapper)
+      .filter(
+        (token): token is Token =>
+          token !== undefined &&
+          token.decimals === 18 &&
+          !aliasesCrv &&
+          ![...gaugeRewardTokens, zeroAddress, NETWORK_TOKEN, aliasesCrv].some((rewardToken) =>
+            isAddressEqual(rewardToken as Address, token.address as Address),
+          ),
+      )
+      .map((token) => ({
+        chain: network?.networkId ?? '',
```

Comment:
this is repeated code

Reply:
My bad, there were multiple instances so I lost track.

For the mapping of 'old' token objects to the TokenOption type required by the token selector feature I've written a small utility function that, given a network, converts it into such:
```ts
export const toTokenOption =
  (chain?: string) =>
  (token: Token): TokenOption => ({
    chain,
    address: token.address as Address,
    symbol: token.symbol,
    volume: token.volume,
  })
```
Example usage:
```ts
tokens.map(toTokenOption(network?.networkId))
```

The `chain`, `label` and `volume` properties in the token option type itself have also been made optional:
```ts
export type TokenOption = {
  address: Address
  symbol: string
  chain?: string
  label?: string
  volume?: number
}
```